infra/modules/providers/azure/app-service/main.tf

```diff
@@ -32,40 +42,25 @@ resource "azurerm_app_service" "appsvc" {
     DOCKER_REGISTRY_SERVER_PASSWORD     = var.docker_registry_server_password
     APPINSIGHTS_INSTRUMENTATIONKEY      = var.app_insights_instrumentation_key
     KEYVAULT_URI                        = var.vault_uri
-    DOCKER_ENABLE_CI                    = var.docker_enable_ci
   }
 
   site_config {
-    linux_fx_version     = format("DOCKER|%s/%s", var.docker_registry_server_url, local.app_configs[count.index].image)
+    linux_fx_version     = local.app_linux_fx_versions[count.index]
     always_on            = var.site_config_always_on
     virtual_network_name = var.vnet_name
   }
 
   identity {
     type = "SystemAssigned"
   }
-}
-
-resource "null_resource" "acr_webhook_creation" {
-  count      = var.docker_enable_ci == true && var.uses_acr ? length(local.app_names) : 0
-  depends_on = [azurerm_app_service.appsvc]
-
-  triggers = {
-    images_to_deploy = "${join(",", [for config in local.app_configs : config.image])}"
-    uses_acr         = var.uses_acr
-  }
-
-  provisioner "local-exec" {
-    command = "az acr webhook create --registry \"$ACRNAME\" --name \"$APPNAME$WEBHOOKNAME\" --actions push --uri $(az webapp deployment container show-cd-url -n $APPNAME_URL -g $APPSVCNAME --query CI_CD_URL -o tsv)"
-
-    environment = {
-      ACRNAME     = var.azure_container_registry_name
-      APPNAME     = replace(lower("${var.app_service_name_prefix}${local.app_names[count.index]}"), "-", "")
-      APPNAME_URL = "${var.app_service_name_prefix}-${local.app_names[count.index]}"
-      WEBHOOKNAME = local.acr_webhook_name
-      APPSVCNAME  = data.azurerm_resource_group.appsvc.name
-    }
 
+  lifecycle {
+    # This stanza will prevent terraform from reverting changes to the application container settings.
+    # These settings are how application teams deploy new containers to the app service and should not
+    # be overridden by Terraform deployments.
+    ignore_changes = [
+      "site_config[0].linux_fx_version"
```

Comment:
so this will set the fx_version on the first setup, then ignore this setting going forward?

Reply:
Yeah thats the long & short of it. If you specify an actual image to deploy it will configure that (on the first time only, since updates are ignored). If you don't, you get an "empty" app svc for containers.